syntax = "proto3";

package template_game;

// 遊戲流程狀態
enum EFlowName {
    FLOW_1 = 0;
    FLOW_2 = 1;
    FLOW_3 = 2;
    FLOW_4 = 3;
    FLOW_5 = 4;
    FLOW_6 = 5;
    FLOW_MAX = 6;
}

// 選項1類型
enum EOption1 {
    OPTION1_1 = 0;
    OPTION1_2 = 1;
    OPTION1_3 = 2;
}

// 選項2類型
enum EOption2 {
    OPTION2_1 = 0;
    OPTION2_2 = 1;
    OPTION2_3 = 2;
    OPTION2_4 = 3;
    OPTION2_MAX = 4;
}

// 完整遊戲狀態訊息
message GameFlowData {
    EFlowName current_flow_state = 1;
    EOption1 selected_option1 = 2;
    EOption2 selected_option2 = 3;
    int64 timestamp = 4;
}

// 遊戲流程控制訊息
message GameFlowControl {
    EFlowName target_state = 1;
    map<string, string> parameters = 2;
    int64 timestamp = 3;
}

// 遊戲流程轉換訊息
message GameFlowTransition {
    EFlowName from_state = 1;
    EFlowName to_state = 2;
    string transition_reason = 3;
    int64 timestamp = 4;
}
